specification/hybridkubernetes/resource-manager/Microsoft.Kubernetes/preview/2024-07-01-preview/connectedClusters.json

```diff
@@ -0,0 +1,1116 @@
+          "description": "Name of the connection",
+          "type": "string",
+          "readOnly": true
+        },
+        "relay": {
+          "description": "Name of the relay.",
+          "type": "string",
+          "readOnly": true
+        },
+        "token": {
+          "description": "Sender access token",
+          "type": "string",
+          "readOnly": true
+        }
+      }
+    },
+    "AadProfile": {
+      "type": "object",
+      "description": "AAD Profile specifies attributes for Azure Active Directory integration.",
+      "properties": {
+        "enableAzureRBAC": {
+          "type": "boolean",
+          "description": "Whether to enable Azure RBAC for Kubernetes authorization."
+        },
+        "adminGroupObjectIDs": {
+          "type": "array",
+          "items": {
+            "type": "string"
+          },
+          "description": "The list of AAD group object IDs that will have admin role of the cluster."
+        },
+        "tenantID": {
+          "type": "string",
+          "description": "The AAD tenant ID to use for authentication. If not specified, will use the tenant of the deployment subscription."
+        }
+      }
+    },
+    "SecurityProfile": {
+      "type": "object",
+      "description": "Security Profile specifies attributes for cluster security features.",
+      "properties": {
+        "workloadIdentity": {
+          "type": "object",
+          "description": "The workload identity feature webhook.",
+          "properties": {
+            "enabled": {
+              "type": "boolean",
+              "description": "Whether to enable or disable the workload identity Webhook"
+            }
+          }
+        }
+      }
+    },
+    "OidcIssuerProfile": {
+      "type": "object",
+      "description": "OIDC Issuer Profile specifies attributes for workload identity integration.",
+      "properties": {
+        "enabled": {
+          "type": "boolean",
+          "description": "Whether to enable oidc issuer for workload identity integration."
+        },
+        "issuerUrl": {
+          "type": "string",
+          "description": "The issuer url for hybrid clusters connected to Arc used for the workload identity feature.",
+          "readOnly": true
+        },
+        "selfHostedIssuerUrl": {
+          "type": "string",
+          "description": "The issuer url for public cloud clusters - AKS, EKS, GKE - used for the workload identity feature."
+        }
+      }
+    },
+    "ConfigurationSettings": {
+      "type": "object",
+      "properties": {
+        "feature": {
+          "type": "string",
+          "description": "Specifies the name of the feature for the configuration setting."
+        },
+        "settings": {
+          "type": "object",
+          "description": "The configuration settings for the feature that do not contain any sensitive or secret information.",
+          "x-nullable": true,
+          "additionalProperties": {
```

Comment:
>additionalProperties [](http://example.com/codeflow?start=11&length=20)

This is an ARM anti-pattern and not generally allowed. Please schematize this object (preferred), or use an array of key value pairs if the properties aren't known ahead of time. If your RP doesn't use or validate these properties at all (i.e. they are just passed through to another API), this pattern is allowed.

Reply:
We have separated the values that we are using in our RP by introducing gateway parameter, making it a first class property. The other values that will be sent in 'arcAgentryConfigurations'(additional property) param will be forwarded to another service.